File: patterns/proxy-item/rules-python/proxy_item_garage.py
from openhab import rule, Registry
from openhab.triggers import ItemCommandTrigger


@rule(triggers=[
    ItemCommandTrigger("Large_Garagedoor_Opener"),
    ItemCommandTrigger("Small_Garagedoor_Opener"),
])
class GaragentorController:
    def execute(self, module, input):
        event = input.get("event")
        if not event:
            return

        if (str(Registry.getItem("GarageControllerComputer").getState()) != "ON"
                or str(Registry.getItem("GarageControllerService").getState()) != "ON"):
            Registry.getItem("AlertItem").sendCommand("Garagentor-Controller offline!")

        item_name = event.getItemName()
        command = str(event.getItemCommand())
        linked_item_name = item_name + "_Linked"
        Registry.getItem(linked_item_name).sendCommand(command)
        self.logger.info(item_name + " -> " + linked_item_name + ": " + command)

File: patterns/proxy-item/rules-js/proxy-item-garage.js
rules.JSRule({
  name: "Garagentor Controller",
  triggers: [
    triggers.ItemCommandTrigger("Large_Garagedoor_Opener"),
    triggers.ItemCommandTrigger("Small_Garagedoor_Opener")
  ],
  execute: (event) => {
    if (items.getItem("GarageControllerComputer").state !== "ON" ||
        items.getItem("GarageControllerService").state !== "ON") {
      items.getItem("AlertItem").sendCommand("Garagentor-Controller offline!");
    }
    items.getItem(event.itemName + "_Linked").sendCommand(event.receivedCommand);
    console.log(event.itemName + " -> " + event.itemName + "_Linked: " + event.receivedCommand);
  }
});
